ceph/ceph_admin/helper.py

```diff
@@ -831,6 +832,57 @@ def validate_log_file_after_enable(cls):
     return True
 
 
+def validate_log_rotate(cls):
+    """
+    Verify logrotation of cephadm.log file
+    Args:
+        cls: cephadm instance object
+
+    Returns:
+        boolean
+    """
+    # step 1: check for the files in current log directory
+    out = cls.installer.exec_command(cmd="ls /var/log/ceph", sudo=True)
+    files = re.split("\n| ", out[0])
+    logFileExists = [f for f in files if f == "cephadm.log"]
+    if not logFileExists:
+        LOG.error("Log rotation failed: Logs not present in /var/log/ceph folder")
+        return False
+
+    # step 2: apply logrotate on cephadm.log
+    out = cls.installer.exec_command(
+        cmd="logrotate --force /etc/logrotate.d/cephadm", sudo=True
+    )
+    # if logrotate doesn't result in logs being zipped return False
+    out = cls.installer.exec_command(cmd="ls /var/log/ceph", sudo=True)
+    files = re.split("\n| ", out[0])
+    logFileExists = [f for f in files if f == "cephadm.log"]
+    if logFileExists:
+        LOG.error("Log rotation failed while enabling rotate")
+        return False
+
+    # step 3: perform any operation that would result in cephadm log be written
+    out, _ = cls.shell(args=["ceph", "orch", "ps", "--format", "json"])
+    daemons = json.loads(out)
+    daemon = [d for d in daemons if "node-exporter" in d["daemon_name"]][0]
+    out, _ = cls.shell(
+        args=["ceph", "orch", "daemon", "redeploy", daemon["daemon_name"]]
+    )
+
+    # step 4: check new cephadm.log file is created
+    out = cls.installer.exec_command(cmd="ls /var/log/ceph", sudo=True)
+    # if new logs are not created return False
+    files = re.split("\n| ", out[0])
+    logFileExists = [f for f in files if f == "cephadm.log"]
```

Comment:
Suggestion : Since this check is been used multiple places, can we make it as a method and optimize the code.

Reply:
Done. I am reusing an already existing method.